src/atlas/grid/detail/grid/CubedSphere.h

```diff
@@ -125,7 +125,9 @@ class CubedSphere : public Grid {
             i_                         = ijt[0];
             j_                         = ijt[1];
             t_                         = ijt[2];
-            compute_point( i_, j_, t_, point_ );
+            if ( grid_.inGrid( i_, j_, t_ ) ) {
+                compute_point( i_, j_, t_, point_ );
+            }
```

Comment:
I found when testing the FV3 cell-center grid that `compute_point` was getting called here for a point not within the grid for panel 5 (the last panel), eventually causing an `ATLAS_ASSERT` to be triggered in `FV3CubedSphereTiles::tileFromXY`. This change fixes the problem, although perhaps someone with more knowledge of the iterator code has a different way they prefer to do it.

Reply:
I am surprised by the need for this.  I does suggest that there is a bug somewhere else. What is the value of the offending xy?

Reply:
Putting an `else` onto the `if` at line 128 that outputs the value of "bad" points produces the following two lines multiple times :
```
Out-of-grid point in CubedSphereIterator::operator++. i_: 2, j_: 2, t_: 5
Out-of-grid point in CubedSphereIterator::operator++. i_: 2, j_: 2, t_: 5
```
I believe `i_` and `j_` should always be either 0 or 1 since we are testing a 2x2 cell-centered grid. For some reason on tile 5 only they are advancing to 2.

Reply:
2,2,5 is the value of the .end() of the iterator. It is the index beyond the points that have values. In the current system it should have an xy but not a lonlat.